### PR TITLE
Add Acic data generating processes + raw data

### PR DESCRIPTION
**Pull request by theo-s**
This includes raw and processed data from the four ACIC data competitions from 2016, 2017, 2018, and 2019. What I have done for all of the data sets is include treatment as a covariate, and then take only the Y corresponding to the potential outcome from the treatment assigned to each unit. So essentially now it is just another feature that contributes to the DGP. If you want to check out how the data processing was done, see `code/generate_acic_data.R`.

All of the processed data sets are saved in `data/` as `ACICX.csv` with X denoting the year. In each cleaned data, there is the true outcome, called `Ytrue` , and the noised outcome, called `Y` . I've also added a function to `bart_stability.R` for pulling training and test sets from the cleaned data.

The data sets have the following dimensions:

1. ACIC 2016: 4802 X 58
2. ACIC 2017: 4302 X 58
3. ACIC 2018: 10391 X 12
4. ACIC 2019: 625 X 23

### CODE REVIEW

File: code/bart_stability.R [hunk 1]
```diff
@@ -54,6 +54,35 @@ get_data <- function(dgp,n, q, rho=0.05, seed=0){
...
+    stop("Not a supported dgp")
+  }
+
+  data <- read.csv(paste0("data/ACIC",dgp,".csv"))
+
+  if (nrow(data) < train_n+test_n) {
+    stop("N is too large for the current dgp")
+  }
+
+  data_train <- data[1:train_n,-which(colnames(data) == "Ytrue")]
```
Comment (OmerRonen) [inline code/bart_stability.R:74]: @theo-s why not take the test set as a random? Looking at the data it seems like similar outcomes are clustered together

File: code/bart_stability.R [hunk 2]
```diff
@@ -54,6 +54,35 @@ get_data <- function(dgp,n, q, rho=0.05, seed=0){
...
+  data_test <- data[(train_n+1):(train_n+200),-which(colnames(data) == "Y")]
+
+  colnames(data_train)[which(colnames(data_train) == "Y")] <- "y"
```
Comment (OmerRonen) [inline code/bart_stability.R:77]: The columns names create a bug with bart_fit, could you please make sure the columns names are the same for train and test? Also could you please check that the CLI works using acic arguments?

File: code/bart_stability.R [hunk 3]
```diff
@@ -30,13 +30,13 @@ get.features <- function(data){
...
+
   }
-  
+
   return(list(X=X, y=y))
-  
+
 }
 
 get_data <- function(dgp,n, q, rho=0.05, seed=0){
```
Comment (OmerRonen) [inline code/bart_stability.R:42]: Please generalize this function to include acic dgps, you can set a default value for q if that is a problem

**Review by OmerRonen — CHANGES_REQUESTED** [review]
@theo-s great work! thanks for doing this so fast, I left a few comments. The main issue is to validate that the CLI works with the new dgps

**Comment by theo-s** [comment]
In the last commit I changed the version of the ACIC 2019 data to use one with continuous outcomes rather than binary outcomes as this was giving BART some problems when it was asked to train on binary features and then calculate RMSE with probabilities. It makes more sense to just use the continuous outcomes. Unfortunately this version only has 500 observations, so will probably be pretty small for any analysis

**Comment by theo-s** [comment]
The CLI is now working as well, although given the size of the data, the range of `n` used in `bart_stablity.R` should be modified before running these dgps

**Pull request merged**